### PR TITLE
Add interactive Code Builder docs page

### PR DESCRIPTION
**Pull request by kingjr**
## Summary

Adds an interactive **Code Builder** page (`docs/neuralset/code_builder.rst`) that replaces the old hand-written quickstart / encoding-decoding pages. Readers pick along six axes — **Neuro × Stim × Task × Model × Config × Compute** — and a complete, ready-to-run NeuralSet script (plus matching `pip install` lines) regenerates live as they click.

A condensed mini-builder is embedded on the NeuralSet landing page (`docs/neuralset/index.rst`) so the first-impression UX is "pick three things, copy a working script".

### Surface

| Axis | Options |
|---|---|
| Neuro | MEG, EEG, iEEG, EMG, fMRI, Spike, fNIRS |
| Stim | Image, Audio, Text, Class, Video |
| Task | Load data, Decoding, Encoding |
| Model | sklearn Ridge, PyTorch (mini-batch `nn.Linear` + `optim.Adam`) |
| Config | kwargs, inline YAML (`exca.TaskInfra`) |
| Compute | local cache, Slurm |

### Real datasets vs synthetic fallback

When `(neuro, stim)` maps to a public dataset shipped through `neuralfetch`, the script targets that study; otherwise it falls back to a new bundled multi-modal synthetic study `ExampleMultiModal` so every combination is runnable end-to-end.

| (neuro, stim) | Study |
|---|---|
| fmri, image | Allen2022MassiveSample |
| fmri, text  | Li2022PetitSample |
| fmri, audio | Bel2026PetitListenSample |
| meg, text   | Bel2026PetitListenSample |
| meg, image  | Grootswagers2022HumanSample |
| meg, classification | Mne2013Sample |
| anything else | ExampleMultiModal (synthetic, ~2 MB) |

`pip install` lines are split into framework+extractor deps and per-study deps so users see exactly what each axis pulls in. Real studies pin a few extractor kwargs (e.g. `allow_maxshield=True` for Bel's MaxShield-recorded MEG, `language="english"` / `"french"` for `SpacyEmbedding` on Li / Bel).

### Architecture

- `docs/_data/code-builder-data.yaml` — single declarative source of truth for axes, options, extractor kwargs, and `(neuro, stim) → study` mappings.
- `docs/_data/build_data.py` — converts the YAML to `_static/code-builder-data.js` during the Sphinx build (wired in `docs/conf.py` via `builder-inited`).
- `docs/_static/code-builder.js` — single client-side renderer (factory `createRenderer(DATA, sel)`), reused by both the browser and the Node fixture generator.
- `docs/_static/code-highlight.js` — small Pygments-style highlighter shared with `code-selector.js`.
- `docs/_static/custom.css` — sticky 2-column grid + collapsible per-axis menus.
- `docs/_data/render_fixtures.mjs` — Node script that imports the renderer factory and emits 16 axis-pinned golden snapshots under `docs/_data/code_builder_fixtures/` (one `script.py` + `install.sh` pair per combo). Replaces a previous Python mirror of the renderer to eliminate drift.
- `neuralset-repo/neuralset/events/testing/examplemultimodal.py` — `ExampleMultiModal` synthetic study covering every neuro × stim combination, including hand-rolled SNIRF (fNIRS) and MP4 (video). Events deliberately skip caching since their absolute filepaths embed `self.path`.

### Tests (`docs/test_code_builder.py`)

| Layer | Coverage | Cost |
|---|---|---|
| L0 | YAML schema, extractor-class resolution, generated-JS-in-sync, fixtures-present, optional fixtures-in-sync (`CB_FIXTURES_REGEN_CHECK=1`) | always runs |
| L1 | `ast.parse` every fixture's `script.py` | always runs |
| L2 | Per-axis pydantic instantiation against real `neuralset.extractors` | always runs |
| L3 | End-to-end exec of every fixture in a clean subprocess (synthetic-only; real-data combos are skipped) | `--slow` only |

```
$ pytest docs/test_code_builder.py
36 passed, 3 skipped in 7.01s
```

Fixture set is deliberately tiny (16 axis-pinned combos) instead of the full 800+ Cartesian product — keeps CI fast while still exercising every option on every axis at least once.

### Cache API

The renderer emits `Study(infra={"folder": CACHE})`. `Study.model_post_init` (already on `main` post-#37) auto-propagates `infra.folder → infra_timelines.folder`, so a single `infra` covers both the merged-events DataFrame cache and the per-timeline cache.

### Minor housekeeping

- `facebook/dinov2-small` added to `neuralset-repo/neuralset/extractors/data/huggingface-repos.txt` (default image extractor).
- `slow` marker registered in `neuralset-repo/conftest.py`; `docs/conftest.py` gains `--slow` to gate L3.
- `pyproject.toml`: `[tool.ruff].exclude` adds `docs/_data/code_builder_fixtures` (must stay byte-identical to the JS renderer output).
- Old `docs/test_code_selector.py` removed (replaced by Code Builder tests).

## Test plan

- [x] `pytest docs/test_code_builder.py` passes (36 / 3 skipped).
- [x] `pytest neuralset-repo/neuralset/events/test_study.py::test_study_caching` passes (validates the `infra → infra_timelines` propagation we now rely on).
- [x] `make -C docs html` builds without Code Builder warnings.
- [ ] CI (Sphinx + ruff + codespell + tests) passes on this branch.
- [ ] Manual smoke: load `code_builder.html`, click through each axis option, confirm both kwargs and YAML render cleanly.

Made with [Cursor](https://cursor.com)

### CODE REVIEW

**Comment by kingjr** [comment]
@BelCorentin if you want to give it a try?

File: neuralset-repo/neuralset/events/testing/examplemultimodal.py [hunk 1]
```diff
@@ -0,0 +1,362 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
```
Comment (kingjr) [inline neuralset-repo/neuralset/events/testing/examplemultimodal.py:1]: @jrapin to replace the combinatorial fake studies

File: neuralset-repo/neuralset/extractors/data/huggingface-repos.txt [hunk 2]
```diff
@@ -38,6 +38,7 @@ facebook/detr-resnet-50
 facebook/dinov2-base
 facebook/dinov2-giant
 facebook/dinov2-large
+facebook/dinov2-small
```
Comment (kingjr) [inline neuralset-repo/neuralset/extractors/data/huggingface-repos.txt:41]: not sure why we didn't have this one
